src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -963,10 +977,10 @@ static ImmutableArray<Symbol> membersToBeInitialized(NamedTypeSymbol containingT
                                 => ImmutableArray<Symbol>.Empty,
 
                             (includeAllMembers: false, includeCurrentTypeRequiredMembers: true, includeBaseRequiredMembers: false)
-                                => containingType.GetMembersUnordered().SelectAsArray(predicate: SymbolExtensions.IsRequired, selector: getFieldSymbolToBeInitialized),
+                                => containingType.GetMembersUnordered().SelectManyAsArray(predicate: SymbolExtensions.IsRequired, selector: getAllMembersToBeDefaulted),
 
                             (includeAllMembers: false, includeCurrentTypeRequiredMembers: true, includeBaseRequiredMembers: true)
-                                => containingType.AllRequiredMembers.SelectAsArray(static kvp => getFieldSymbolToBeInitialized(kvp.Value)),
+                                => containingType.AllRequiredMembers.SelectManyAsArray(static kvp => getAllMembersToBeDefaulted(kvp.Value)),
 
                             (includeAllMembers: true, includeCurrentTypeRequiredMembers: _, includeBaseRequiredMembers: false)
                                 => containingType.GetMembersUnordered().SelectAsArray(getFieldSymbolToBeInitialized),
```

Comment:
>getFieldSymbolToBeInitialized [](http://example.com/codeflow?start=86&length=29)

Should this call site be changed too?

Reply:
No, there's no new member that `MemberNotNull` could introduce here: we're already including all members in the type, and `MemberNotNull` is only valid on members in the current type (referencing a member from a base type is a warning and has no effect).